modules/outbound-command-event-handler/src/domain/bulk_transaction_agg.ts

```diff
@@ -0,0 +1,154 @@
+ * Modusbox
+ - Vijay Kumar Guthi <vijaya.guthi@modusbox.com>
+ --------------
+ ******/
+
+'use strict';
+
+import { ILogger } from '@mojaloop/logging-bc-public-types-lib';
+import { BaseAggregate, IEntityStateRepository } from '@mojaloop/sdk-scheme-adapter-private-shared-lib';
+import { BulkTransactionEntity, BulkTransactionState } from './bulk_transaction_entity';
+import {
+    IndividualTransferEntity,
+    IndividualTransferState,
+} from './individual_transfer_entity';
+import { IBulkTransactionEntityRepo } from '../types';
+
+export class BulkTransactionAgg extends BaseAggregate<BulkTransactionEntity, BulkTransactionState> {
+    // TODO: These counts can be part of bulk transaction entity?
+    // private _partyLookupTotalCount?: number;
+    // private _partyLookupSuccessCount?: number;
+    // private _partyLookupFailedCount?: number;
+    // // NEXT_STORY
+    // private _bulkBatches: Array<BulkBatchState>; // Create bulk batch entity
+    // private _bulkQuotesTotalCount: number;
+    // private _bulkQuotesSuccessCount: number;
+    // private _bulkQuotesFailedCount: number;
+    // private _bulkTransfersTotalCount: number;
+    // private _bulkTransfersSuccessCount: number;
+    // private _bulkTransfersFailedCount: number;
+
+    // constructor(
+    //   bulkTransactionEntity: BulkTransactionEntity,
+    //   entityStateRepo: IEntityStateRepository<BulkTransactionState>,
+    //   logger: ILogger
+    // ) {
+    //     super(bulkTransactionEntity, entityStateRepo, logger);
+    // }
+
+    static async CreateFromRequest(
+        /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+        request: any,
+        entityStateRepo: IEntityStateRepository<BulkTransactionState>,
+        logger: ILogger,
+    ): Promise<BulkTransactionAgg> {
+    // Create root entity
+        const bulkTransactionEntity = BulkTransactionEntity.CreateFromRequest(request);
+        // Create the aggregate
+        const agg = new BulkTransactionAgg(bulkTransactionEntity, entityStateRepo, logger);
+        // Persist in the rep
+        await agg.store();
+        // Create individualTransfer entities
+        if(Array.isArray(request?.individualTransfers)) {
+            // TODO: limit the number of concurrently created promises to avoid nodejs high memory consumption
+            await Promise.all(request.individualTransfers.map((individualTransfer: any) =>
+                agg.addIndividualTransferEntity(IndividualTransferEntity.CreateFromRequest(individualTransfer))));
+        }
+        // Return the aggregate
+        return agg;
+    }
+
+    static async CreateFromRepo(
+        id: string,
+        entityStateRepo: IEntityStateRepository<BulkTransactionState>,
+        logger: ILogger,
+    ): Promise<BulkTransactionAgg> {
+    // Get the root entity state from repo
+        let state: BulkTransactionState | null;
+        try {
+            state = await entityStateRepo.load(id);
```

Comment:
Referring to the base-agg comment about why we should NOT pass `bulkTransactionEntity` as part of the constructor....and expanding on the above comment...

1. The Aggregator is tightly coupled to its state-repo (as it should be)
2. The Aggregator should know how to load its own state, and how to save its own state...and its perfectly valid to have "NO/empty state" (which is useful for duplicate checks, etc)
3. Factory functions should be considered "helpers", and should not be relied on <-- here we now have the entityStateRepo being loaded externally to the instantiated aggregator.  So we have thus coupled the Repo to both the Aggregator and the Factory function, which is bad practice.